rfc/rfc018-openid-credential-issuance.md

```diff
@@ -0,0 +1,144 @@
+If the wallet has retrieved the credential, it can respond with `status` set to `credential_received`.
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "status": "credential_received"
+}
+```
+
+### 3.2.2 Credential Offer Error Response
+
+This specification extends the specification with the following error responses a wallet MUST support when receiving a credential offer:
+
+- `invalid_request`: the offer is missing or invalid.
+- `invalid_grant`: the offer does not contain a grant the wallet could use to retrieve the credential.
+
+It MUST respond with HTTP status `400 Bad Request` and a response body containing the error code.
+
+Below is a non-normative example of such a response:
+
+```http
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Cache-Control: no-store
+
+{
+  "error": "invalid_grant"
+}
+```
+
+## 4. Client Metadata Discovery using DID documents
+
+To offer a credential the issuer needs to discover the wallet's credential offer endpoint, which is registered in the client metadata.
+
+To allow a (Nuts) wallet to be discovered, the wallet MUST be represented by a resolvable DID.
+The DID document MUST contain a `node-http-services-baseurl` service, which contains the base URL for node-to-node HTTP services.
+The issuer appends the following path to it to resolve the client metadata: `/n2n/identity/<did>/.well-known/openid-credential-wallet`,
+replacing `<did>` with the DID of the wallet holder.
+
+## 5. Security
+
+### 5.1 Credential Issuer Trust
+
+In server-to-server issuance of credentials, there's no user approving the credential offer before requesting it and loading it into the holder's wallet.
+Therefore, the holder must make a decision whether to trust the credential issuer.
+Trust SHOULD be derived from the TLS trust anchors referred to by section 5.2 (Transport Security).
+
+### 5.2 Transport Security
+
+To prevent against eavesdropping and man-in-the-middle attacks, all exchanges MUST be done over HTTPS.
+Clients should refuse to connect to plain HTTP endpoints.
+Refer to [RFC008 Certificate Structure](rfc008-certificate-structure.md) for requirements regarding these certificates and which Certificate Authorities should be accepted.
+
+## Appendix A: Design decisions
+
+### A.1 No PIN requirement
+
+OpenID4VCI specifies a PIN to mitigate credential stealing,
+e.g. someone looking over the shoulder of the intended holder standing at a terminal,
+quickly scanning the QR code and stealing the resulting credential.
+Since this specification describes server-to-server flows, there is no user involved from who the QR code can be stolen
+or coerced into giving the QR code.
+To prevent eavesdropping, HTTPS MUST be used to protect communication (see Appendix B.1).
+
+### A.2 Trust issuer based on TLS certificates
```

Comment:
Do we do this currently?

Reply:
I think I tried to convey that we only trust issuers from trusted certificates, which is defined by the truststore. And since identifier URLs are HTTPS URLs, they're trusted if the connection can be made.